### ISSUE TITLE
Add morphology support to glossary and search

### ISSUE THREAD

**Issue by tariver** [WeblateOrg/weblate#3023]
**Is your feature request related to a problem? Please describe.**
Base language of my project is Russian so my glossary is Russian>Other language
But some terms are not showed of they are in different case. As an example:
Our project frequently uses the word "карта" (map). So I made an glossary entry for term "карта".
If the source string contains this term in nominative case (карт**а**) the glossary shows it without problem. But if the source contains this word in genitive case (карт**ы**) the term is not shown in the glossary tab.
Although sometimes it works. For example glossary term "справочник" (reference) shows up if the source contains plural form "справочник**и**".
The same goes to the "Search" functions - it doesn't search for all the forms of a word.

**Describe the solution you'd like**
There is an open source spell checker [Hunspell](http://hunspell.github.io) which has a lot of dictionaries created for it (for example [here](https://github.com/wooorm/dictionaries/tree/master/dictionaries)) and it would be nice to implement it's morphology dictionaries into Weblate's glossary and search functions.

**Describe alternatives you've considered**
An alternative is to input all the word's forms into glossary, but it's a cumbersome task.

**Comment by nijel** [comment]
Right now it might be doable with [Whoosh stemmers](https://github.com/whoosh-community/whoosh/tree/master/src/whoosh/lang/snowball), in the long term better approach will be to rely on fulltext capabilities in PosgtreSQL, see https://github.com/WeblateOrg/weblate/issues/2825.

**Comment by tariver** [comment]
Just to clarify - is there a way to implement this right now or Weblate's code needs to be changed? I had a look at the Weblate files and these stemmer files are present in the whoosh folder.

**Comment by nijel** [comment]
There is probably way to implement it in current code base using Whoosh (though it's currently not used for glossary lookup). However it's probably not that good idea to spend effort on this right now, given that this code is going to be replaced by using PostgreSQL features (see #2825).

**Comment by northantech** [comment]
The same issue applies to Turkish language as well. Just as a user, I'm not very informed of the source code, but I believe agglutinative language support would be awesome in glossary.

**Comment by Cerno-b** [comment]
It even is a problem in English. If you have entered a singular term into the glossary, then plural versions of the word do not find the glossary entry.

**Comment by Cerno-b** [comment]
@nijel Do you think this is going to be addressed? I am currently cleaning up our glossary and there are a lot of plural forms that I would like to transform to singular. If I understand this correctly, then in order to find a glossary term in the sidebar, I would have to add an entry for the singular **and** the plural to the glossary if I want to make sure that the word appears in the sidebar in all cases. Is that correct?

A solution that will probably not help OP, but it could lead to a lot more hits would be to trigger a sidebar term if a substring of the translation text is in the glossary.

That would mean that a glossary entry for "asset" would show up in a text that contains "asset" and "assets". It would probably lead to some false positives, but I think the benefits outweigh the disadvantages.

Having an extra field where the user can add morphological alternatives to a word would probably be a better solution, but I guess that would be a lot more effort as well. :/

**Comment by nijel** [comment]
I don't think substring matching is a way to go, as that would lead to too many false matches.

You would rather not see “map” glossary term when there is “maple” in the source.

**Comment by Cerno-b** [comment]
@nijel I agree that it's not ideal, but as a quick fix, I would probably prefer some FPs than missing out on half of the glossary entries if all plurals will be hidden from the sidebar.

Of course proper morphology would be better, that's why I wondered if this is on the roadmap in the near future.

**Comment by nijel** [comment]
I don't think it's a reasonable thing to do even as a quick fix, as it will produce many false matches. Actually, it would only produce false matches for most of the languages (see Russian example in the initial post).

**Comment by Cerno-b** [comment]
Sure, I'm fine with that, and I'm not really a fan of half-baked solutions. So since proper morphology support sounds to be a lot of effort, do you think the multiple value approach discussed in #7416 is a viable alternative that takes less effort to realize?
